src/Type/Definition/UnionType.php

```diff
@@ -53,7 +55,7 @@ public function __construct(array $config)
 
     public function isPossibleType(Type $type) : bool
     {
-        if (! $type instanceof ObjectType) {
+        if (! ($type instanceof ObjectType && $type instanceof Type)) {
```

Comment:
Why this `$type instanceof ObjectType` is not enough? `ObjectType extends Type` so it seems to me that 2nd part will always be true if 1st part is.

Reply:
That's a good point! I wonder what I was thinking...

Reply:
Simplified in 88db0b3, but I still don't remember what led me to do this 🤔